py/torch_tensorrt/_Input.py

```diff
@@ -364,9 +364,13 @@ def example_tensor(
                     )
 
                 if isinstance(self.shape, dict):
-                    return torch.rand(self.shape[optimization_profile_field]).to(
-                        dtype=self.dtype.to(torch.dtype, use_default=True)
-                    )
+                    if self.dtype == dtype.b:
```

Comment:
Could you provide more details on why this part was changed? I'd like to understand it better.

Reply:
Casting from random [0,1) value to bool type is not efficient as all output is True except 0.
I think random [0,1) < 0.5 should be helpful to create better random bool values